### PR TITLE
Running Quantized ResNet50 via MIGraphX

### PR DESCRIPTION
**Pull request by Rmalavally**
https://ontrack-internal.amd.com/browse/SWDEV-427204

https://ontrack-internal.amd.com/browse/ROCDOC-210

### CODE REVIEW

**Review by emankov — APPROVED** [review]
LGTM

**Review by saadrahim — CHANGES_REQUESTED** [review]
There are two changes included in this PR. They should be separated. I am not sure what the goal is of this yet. Frameworks-integrations should not be the directory name. Directory name needs some thought.

**Comment by Rmalavally** [comment]
> There are two changes included in this PR. They should be separated. I am not sure what the goal is of this yet. Frameworks-integrations should not be the directory name. Directory name needs some thought.

Thanks, Saad. Deleted the ASAN file. 

'Frameworks integrations' will include examples of ROCm applications in PyTorch, Tensorflow, and others. The recommendation came from the MIGraphX team. Please let me know if you have any suggestions. @saadrahim @shivadbhavsar

**Comment by LisaDelaney** [comment]
@Rmalavally the syntax isn't consistent with our style guide. Can you please update it? See: https://rocm.docs.amd.com/en/latest/contribute/contribute-docs.html

File: Frameworks-integrations/MIGraphX/Quantizations.rst
Comment (saadrahim) [inline Frameworks-integrations/MIGraphX/Quantizations.rst]: What is a PTQ?

File: Frameworks-integrations/MIGraphX/Quantizations.rst
Comment (saadrahim) [inline Frameworks-integrations/MIGraphX/Quantizations.rst]: Please conclude this example

File: Frameworks-integrations/MIGraphX/Quantizations.rst
Comment (saadrahim) [inline Frameworks-integrations/MIGraphX/Quantizations.rst]: installation instructions for what?

File: Frameworks-integrations/MIGraphX/Quantizations.rst
Comment (saadrahim) [inline Frameworks-integrations/MIGraphX/Quantizations.rst:1]: Folder can be AI?

File: Frameworks-integrations/MIGraphX/Quantizations.rst
Comment (saadrahim) [inline Frameworks-integrations/MIGraphX/Quantizations.rst]: The meaning of this title does not make sense to me. import lower quantized model, ingest lower quantized model, build lower quantized model?

File: Frameworks-integrations/MIGraphX/Quantizations.rst
Comment (saadrahim) [reply Frameworks-integrations/MIGraphX/Quantizations.rst]: What was the performance gain? I don't want to run to find out, I'd like to know when I am reading.

**Review by saadrahim — CHANGES_REQUESTED** [review]
This needs to be added in the TOC of the main README.md file in the repo.

It needs to be reviewed by the MIGraphX team.

What is the purpose of this example? In general, I am not sure if this is an example or if it should be some form of blog.

**Comment by saadrahim** [comment]
> @Rmalavally the syntax isn't consistent with our style guide. Can you please update it? See: https://rocm.docs.amd.com/en/latest/contribute/contribute-docs.html

This repo does not use RST as it has no generated documentation. Please convert to markdown.

**Comment by Rmalavally** [comment]
> @Rmalavally the syntax isn't consistent with our style guide. Can you please update it? See: https://rocm.docs.amd.com/en/latest/contribute/contribute-docs.html

Does the style guide apply to ROCm examples as well? Please confirm.

File: Frameworks-integrations/MIGraphX/Quantizations.rst
Comment (Rmalavally) [reply Frameworks-integrations/MIGraphX/Quantizations.rst]: This example is reviewed and approved by the MIGraphX team. 

@shivadbhavsar, Could you please provide context?

File: Frameworks-integrations/MIGraphX/Quantizations.rst
Comment (Rmalavally) [reply Frameworks-integrations/MIGraphX/Quantizations.rst:1]: Not sure about AI. Let me get back to you on this.

File: Frameworks-integrations/MIGraphX/Quantizations.rst
Comment (Rmalavally) [reply Frameworks-integrations/MIGraphX/Quantizations.rst]: @LisaDelaney @saadrahim Let's discuss the format for examples. Adding much content will change this from an example to a blog/article. I am ok if we want to convert this to a blog. 

@emankov @shivadbhavsar

**Comment by Rmalavally** [comment]
> This needs to be added in the TOC of the main README.md file in the repo.
> 
> It needs to be reviewed by the MIGraphX team.
> 
> What is the purpose of this example? In general, I am not sure if this is an example or if it should be some form of blog.

This was reviewed and approved by the MIGraphX team.

**Comment by Rmalavally** [comment]
> > @Rmalavally the syntax isn't consistent with our style guide. Can you please update it? See: https://rocm.docs.amd.com/en/latest/contribute/contribute-docs.html
> 
> This repo does not use RST as it has no generated documentation. Please convert to markdown.

I can change it to Markdown, not a problem. 

@LisaDelaney - The Libraries examples have Summary and Prerequisites, while the HIP examples have Description and Key APIs. Which one should we follow for examples? There is no consistency.

Each folder has examples in different styles. My recommendation is to create a simple format (we could finalize on either the Libraries or HIP formats) for all examples.

**Comment by Rmalavally** [comment]
> > @Rmalavally the syntax isn't consistent with our style guide. Can you please update it? See: https://rocm.docs.amd.com/en/latest/contribute/contribute-docs.html
> 
> Does the style guide apply to ROCm examples as well? Please confirm.

Using the contributing guide will make it inconsistent with other examples.

**Comment by Rmalavally** [comment]
> > @Rmalavally the syntax isn't consistent with our style guide. Can you please update it? See: https://rocm.docs.amd.com/en/latest/contribute/contribute-docs.html
> 
> This repo does not use RST as it has no generated documentation. Please convert to markdown.

@saadrahim this is the guideline in our style guide. Please advise. 

>Our documentation includes both Markdown and RST files. We are gradually transitioning existing Markdown to RST in order to more effectively meet our documentation needs. When contributing, RST is preferred; if you must use Markdown, use GitHub-flavored Markdown.

File: Frameworks-integrations/MIGraphX/Quantizations.rst
Comment (Rmalavally) [reply Frameworks-integrations/MIGraphX/Quantizations.rst:1]: Sounds good, @saadrahim. Thanks for the suggestion.

File: Frameworks-integrations/MIGraphX/Quantizations.rst
Comment (Rmalavally) [reply Frameworks-integrations/MIGraphX/Quantizations.rst]: Post Training Quantization. Corrected. @saadrahim

File: Frameworks-integrations/MIGraphX/Quantizations.rst
Comment (Rmalavally) [reply Frameworks-integrations/MIGraphX/Quantizations.rst]: I have replicated the installation instructions in the Readme file and assume any changes to the original instructions will also be reflected here. Otherwise, it could be a maintenance task. Please advise.  @saadrahim @emankov @shivadbhavsar

Comment (shivadbhavsar) [reply Frameworks-integrations/MIGraphX/Quantizations.rst]: Installation instructions for the `torch_migraphx` library (where the link points)

File: Frameworks-integrations/MIGraphX/Quantizations.rst
Comment (shivadbhavsar) [reply Frameworks-integrations/MIGraphX/Quantizations.rst]: "Lower" here is used as a verb. Its a commonly used term in the context of optimizing deeplearning models on specific hardware. It means this step is "lowering" the model from the framework level to the graph compiler level

File: Frameworks-integrations/MIGraphX/Quantizations.rst
Comment (Rmalavally) [reply Frameworks-integrations/MIGraphX/Quantizations.rst]: > What was the performance gain? I don't want to run to find out, I'd like to know when I am reading.

@shivadbhavsar Please advise.

File: Frameworks-integrations/MIGraphX/Quantizations.rst
Comment (Rmalavally) [reply Frameworks-integrations/MIGraphX/Quantizations.rst]: > installation instructions for what?

Updated.

File: AI/MIGraphX/Quantization/Readme.md
Comment (shivadbhavsar) [inline AI/MIGraphX/Quantization/Readme.md:48]: Im not sure if it's a good idea to copy paste the readme from torch_migraphx rather than just link to it

File: AI/MIGraphX/Quantization/Readme.md
Comment (shivadbhavsar) [inline AI/MIGraphX/Quantization/Readme.md]: Even if the decision is to leave in the pasted installation instructions, the content below this line is probably not necessary since the examples in this repo themselves will show the user how to use the provided APIs in this context.

File: AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md
Comment (shivadbhavsar) [inline AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md]: Remove this note, its not relevant anymore.

File: AI/MIGraphX/Quantization/Readme.md
Comment (Rmalavally) [reply AI/MIGraphX/Quantization/Readme.md:48]: Agreed and corrected.

File: AI/MIGraphX/Quantization/Readme.md
Comment (Rmalavally) [reply AI/MIGraphX/Quantization/Readme.md]: @shivadbhavsar I added a link to the Torch_MIGraphX page.

File: AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md
Comment (Rmalavally) [reply AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md]: Removed.

File: AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md
Comment (LisaDelaney) [inline AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md]:
```suggestion
Do a quick test to measure the performance gain from using quantization.
```

**Comment by LisaDelaney** [comment]
@Rmalavally there are a couple titles not in sentence case--mind fixing those?

**Comment by LisaDelaney** [comment]
> > > @Rmalavally the syntax isn't consistent with our style guide. Can you please update it? See: https://rocm.docs.amd.com/en/latest/contribute/contribute-docs.html
> > 
> > 
> > This repo does not use RST as it has no generated documentation. Please convert to markdown.
> 
> @saadrahim this is the guideline in our style guide. Please advise.
> 
> > Our documentation includes both Markdown and RST files. We are gradually transitioning existing Markdown to RST in order to more effectively meet our documentation needs. When contributing, RST is preferred; if you must use Markdown, use GitHub-flavored Markdown.

@Rmalavally if this repo is intended to be viewed mostly from GitHub (which I think it is), markdown makes more sense in this case. If this changes in the future, we can sort it out then.

**Comment by Rmalavally** [comment]
> > > > @Rmalavally the syntax isn't consistent with our style guide. Can you please update it? See: https://rocm.docs.amd.com/en/latest/contribute/contribute-docs.html
> > > 
> > > 
> > > This repo does not use RST as it has no generated documentation. Please convert to markdown.
> > 
> > 
> > @saadrahim this is the guideline in our style guide. Please advise.
> > > Our documentation includes both Markdown and RST files. We are gradually transitioning existing Markdown to RST in order to more effectively meet our documentation needs. When contributing, RST is preferred; if you must use Markdown, use GitHub-flavored Markdown.
> 
> @Rmalavally if this repo is intended to be viewed mostly from GitHub (which I think it is), markdown makes more sense in this case. If this changes in the future, we can sort it out then.

Sounds good. Please add this guideline to the style guide, or I could too.

**Comment by Rmalavally** [comment]
> > > > > @Rmalavally the syntax isn't consistent with our style guide. Can you please update it? See: https://rocm.docs.amd.com/en/latest/contribute/contribute-docs.html
> > > > 
> > > > 
> > > > This repo does not use RST as it has no generated documentation. Please convert to markdown.

Please update the style guide with this detail.
> > > 
> > > 
> > > @saadrahim this is the guideline in our style guide. Please advise.
> > > > Our documentation includes both Markdown and RST files. We are gradually transitioning existing Markdown to RST in order to more effectively meet our documentation needs. When contributing, RST is preferred; if you must use Markdown, use GitHub-flavored Markdown.
> > 
> > 
> > @Rmalavally if this repo is intended to be viewed mostly from GitHub (which I think it is), markdown makes more sense in this case. If this changes in the future, we can sort it out then.
> 
> Sounds good. Please add this guideline to the style guide, or I could too.

File: AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md
Comment (Rmalavally) [reply AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md]: @saadrahim The comments have been resolved. Please let me know if you have any other recommendations. If not, could you please unblock for merging? Thanks so much.

**Comment by LisaDelaney** [comment]
> > > > > > @Rmalavally the syntax isn't consistent with our style guide. Can you please update it? See: https://rocm.docs.amd.com/en/latest/contribute/contribute-docs.html
> > > > > 
> > > > > 
> > > > > This repo does not use RST as it has no generated documentation. Please convert to markdown.
> 
> Please update the style guide with this detail.
> 
> > > > @saadrahim this is the guideline in our style guide. Please advise.
> > > > > Our documentation includes both Markdown and RST files. We are gradually transitioning existing Markdown to RST in order to more effectively meet our documentation needs. When contributing, RST is preferred; if you must use Markdown, use GitHub-flavored Markdown.
> > > 
> > > 
> > > @Rmalavally if this repo is intended to be viewed mostly from GitHub (which I think it is), markdown makes more sense in this case. If this changes in the future, we can sort it out then.
> > 
> > 
> > Sounds good. Please add this guideline to the style guide, or I could too.

This does not need to be added to our style guide. You can make decisions based on context using critical thinking and logical thought. As a tech writer, minor/once-off exceptions are at your discretion; these do not need to clutter up a style guide.

**Comment by Rmalavally** [comment]
> > > > > > > @Rmalavally the syntax isn't consistent with our style guide. Can you please update it? See: https://rocm.docs.amd.com/en/latest/contribute/contribute-docs.html
> > > > > > 
> > > > > > 
> > > > > > This repo does not use RST as it has no generated documentation. Please convert to markdown.
> > 
> > 
> > Please update the style guide with this detail.
> > > > > @saadrahim this is the guideline in our style guide. Please advise.
> > > > > > Our documentation includes both Markdown and RST files. We are gradually transitioning existing Markdown to RST in order to more effectively meet our documentation needs. When contributing, RST is preferred; if you must use Markdown, use GitHub-flavored Markdown.
> > > > 
> > > > 
> > > > @Rmalavally if this repo is intended to be viewed mostly from GitHub (which I think it is), markdown makes more sense in this case. If this changes in the future, we can sort it out then.
> > > 
> > > 
> > > Sounds good. Please add this guideline to the style guide, or I could too.
> 
> This does not need to be added to our style guide. You can make decisions based on context using critical thinking and logical thought. As a tech writer, minor/once-off exceptions are at your discretion; these do not need to clutter up a style guide.

In general, i think good for consistency as it digresses from our standard practices. I am good with all options.

File: AI/MIGraphX/Quantization/Readme.md
Comment (saadrahim) [inline AI/MIGraphX/Quantization/Readme.md]:
```suggestion
Quantization is an optimization tool that is commonly used to accelerate inferencing for deep learning models. There are many flavors of quantization. In the examples here, we show how the ROCm can be used to accelerate inference using [INT8 quantization](https://arxiv.org/pdf/1712.05877.pdf) on pretrained models. 
```

File: AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md
Comment (saadrahim) [inline AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md]: personal forks should not be linked to.

File: AI/MIGraphX/Quantization/Readme.md
Comment (saadrahim) [inline AI/MIGraphX/Quantization/Readme.md]:
```suggestion
The examples in this subdirectory showcase the functionality of the [MIGraphX](https://github.com/ROCmSoftwarePlatform/torch_migraphx/tree/master) library. The examples build on Linux for the ROCm (AMD GPU) backend.
```
 Last sentence is too technical, let's rewrite.  AMD GPU backend term does not need to be used.

**Review by saadrahim — CHANGES_REQUESTED** [review]

File: AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md
Comment (Rmalavally) [reply AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md]: Will update this once we have the approval to merge.

File: AI/MIGraphX/Quantization/Readme.md
Comment (shivadbhavsar) [inline AI/MIGraphX/Quantization/Readme.md]: For the link please change text to Torch-MIGraphX from just MIGraphX. Thats confusing because MIGraphX is its own repo (not the one we are pointing to in this case)

Comment (Rmalavally) [reply AI/MIGraphX/Quantization/Readme.md]: Fixed. Please review the updated Summary section in the Readme file.

File: AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md
Comment (Rmalavally) [reply AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md]: Added. Please review the Readme file in rocm-examples.

File: AI/MIGraphX/Quantization/Readme.md
Comment (LisaDelaney) [inline AI/MIGraphX/Quantization/Readme.md]:
```suggestion
Quantization is an optimization tool that is commonly used to accelerate inferencing for deep learning models. There are many flavors of quantization. In the examples here, we show you how to use ROCm to accelerate inference using [INT8 quantization](https://arxiv.org/pdf/1712.05877.pdf) on pretrained models. 
```

File: AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md
Comment (shivadbhavsar) [inline AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md]: This line can be removed, its not relevant to the model in this example

File: AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md
Comment (shivadbhavsar) [inline AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md]: This can be removed, it will probably just confuse the user. We will need to update this example when/if PyTorch makes this change.

**Review by shivadbhavsar — CHANGES_REQUESTED** [review]

File: AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md
Comment (Rmalavally) [reply AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md]: Fixed

**Review by shivadbhavsar — APPROVED** [review]

File: AI/MIGraphX/Quantization/Readme.md
Comment (causten) [inline AI/MIGraphX/Quantization/Readme.md:49]: The repo moved to https://github.com/ROCm/torch_migraphx/blob/master/README.md

File: AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md
Comment (causten) [inline AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md:10]: Change to the correct org

**Review by causten — CHANGES_REQUESTED** [review]

File: AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md
Comment (Rmalavally) [reply AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md:10]: @causten The content in Readme.md accompanying the example is different from the Readme in the repo. 

We have the following link at the end of the file for installation instructions,

>Refer to the [Torch_MIGraphX](https://github.com/ROCmSoftwarePlatform/torch_migraphx/blob/master/README.md) page for Docker and source installation instructions.

@shivadbhavsar Please confirm.

Comment (Rmalavally) [reply AI/MIGraphX/Quantization/Running Quantized ResNet50 via MIGraphX.md:10]: @causten Let me know if you have any concerns. The Readme with the example provides contextual details about the example. If you are ok with this, can we merge the PR? Please advise.

**Review by saadrahim — APPROVED** [review]

**Pull request merged**
